src/Main.elm

```diff
@@ -222,53 +248,90 @@ view model =
-
-        habitCompleteButton =
-            if completedToday model then
-                div []
-                    [ div
-                        [ class <| primaryButton ++ " text-gray-800"
-                        , onClick RemoveLastHabitEntry
-                        ]
-                        [ Icons.undo purple 36 ]
-                    ]
 
-            else
-                button
-                    [ class <| primaryButton ++ " bg-purple-700 hover:bg-purple-800", onClick CompleteHabit ]
-                    [ Icons.check white 36 ]
+daysCompleted : Model -> Int
+daysCompleted model =
+    timesHabitWasCompleted model.habit model.habitLog
+
+
+habitScreen : Model -> Html Msg
+habitScreen model =
+    let
+        daysCompletedText =
+            String.fromInt (daysCompleted model) ++ " / 21"
     in
-    div []
-        [ div [ class "grid h-screen px-4 py-12" ]
-            [ div [ class "flex flex-col justify-end" ]
-                [ h1
-                    [ classList
-                        [ ( header ++ " overflow-y-scroll max-h-64 mb-10 break-anywhere", True )
-                        , ( "line-through", completedToday model )
-                        ]
-                    , onClick ToggleEditHabit
-                    ]
-                    [ text model.habit ]
-                , div [ class daysText ] [ text daysCompletedText ]
-                ]
-            , div [ class "flex items-end justify-center" ] [ habitCompleteButton ]
+    div [ class "fixed grid h-screen w-screen px-4 py-12 z-10" ]
+        [ div [ class "flex flex-col justify-end" ]
+            [ habitTextView model
+            , div [ class daysText ] [ text daysCompletedText ]
             ]
-        , editHabitModal model
+        , div [ class "flex items-end justify-center" ] [ habitCompleteButton model ]
         , completedTodayText model
         ]
 
 
+progressBar : Model -> Html Msg
+progressBar model =
+    let
+        totalDays =
+            21
+
+        completedDays =
+            toFloat <| daysCompleted model
+
+        progressBarHeight =
+            (completedDays / totalDays) * model.viewportSize.height
+
+        progressBarHeightString =
+            (String.fromInt <| round progressBarHeight) ++ "px"
+    in
+    div
+        [ class "fixed bottom-0 left-0 w-full z-0 bg-purple-100"
+        , style "height" progressBarHeightString
+        ]
+        []
+
+
+habitTextView : Model -> Html Msg
+habitTextView model =
+    h1
+        [ classList
+            [ ( header ++ " overflow-y-scroll max-h-64 mb-10 break-anywhere", True )
+            , ( "line-through", completedToday model )
+            ]
+        , onClick ToggleEditHabit
+        ]
+        [ text model.habit ]
+
+
+habitCompleteButton : Model -> Html Msg
+habitCompleteButton model =
+    if completedToday model then
+        div []
+            [ button
+                [ class <| primaryButton ++ " text-gray-800"
+                , onClick RemoveLastHabitEntry
```

Comment:
`RemoveLastHabitEntry` concerns me slightly as being a spot for a lot of edge case bugs. For example, what happens if a user
- completes HabitA
- then changes the habit to HabitB
- completes HabitB
- changes back to HabitA
- and then undoes complete Habit
?
would HabitA be removed or HabitB? (I think HabitB in this scenario) what should be removed?

and im sure theres a lot of others.

I think to add this behavior properly we may need to rethink the data structures we are using for the `HabitHistory`

Reply:
Ahhhh yes lmk if you'd like to 🍐 on this tonight!